rust/aspects.bzl

```diff
@@ -125,14 +125,20 @@ def _generate_rust_gencode(
         ),
     )
 
+    proto_root = proto_info.proto_source_root
+    # Handles virtual import cases
+    if proto_root.startswith(ctx.bin_dir.path):
+        proto_root = proto_root[len(ctx.bin_dir.path) + 1:]
+
+
     proto_common.compile(
         actions = ctx.actions,
         proto_info = proto_info,
         additional_inputs = depset(direct = [crate_mapping]),
         additional_args = additional_args,
         generated_files = rs_outputs + cc_outputs,
         proto_lang_toolchain_info = proto_lang_toolchain,
-        plugin_output = ctx.bin_dir.path,
+        plugin_output = ctx.bin_dir.path + "/" + proto_root
```

Comment:
`plugin_output` is deprecated: https://github.com/protocolbuffers/protobuf/blob/ebbb1a465fedf69f7cb450c9133d9b1a49678a68/bazel/common/proto_common.bzl#L142-L145

I think it would be better to just remove it if possible. Does that work?

Reply:
Sounds good to me! Do you want to handle it or would you like me to?

Reply:
Please include the change in the PR

Reply:
Done!